### PR TITLE
feat: Add Native Token Limit Module Support

### PR DESCRIPTION
**Pull request by Zer0dot**
# Pull Request Checklist

- [x] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [x] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [x] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR focuses on updating the handling of `entityId` types from `number` to `bigint`, introducing the `nativeTokenLimitModule`, and enhancing testing scenarios for module installations and validations.

### Detailed summary
- Changed `entityId` type from `number` to `bigint` in `InvalidEntityIdError`.
- Commented out `throw new EntityIdOverrideError()` in `client.ts`.
- Introduced `nativeTokenLimitModule` with `encodeOnInstallData` and `encodeOnUninstallData` methods.
- Updated tests to include installation and validation of `nativeTokenLimitModule`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: account-kit/smart-contracts/src/ma-v2/client/client.test.ts [hunk 1]
```diff
@@ -258,6 +259,105 @@ describe("MA v2 Tests", async () => {
...
+        uo: {
+          target: target,
+          value: parseEther("0.2"), // Try to send 0.2 ETH, above the 0.1 limit
+          data: "0x",
+        },
+      })
+    ).rejects.toThrowError();
+
+    // Try to send less than the limit - should succeed
+    const sendResult = await provider.sendUserOperation({
```
Comment (moldy530) [inline account-kit/smart-contracts/src/ma-v2/client/client.test.ts:319]: ⚠️ **[eslint]** <[@typescript-eslint/no-unused-vars](https://typescript-eslint.io/rules/no-unused-vars)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'sendResult' is assigned a value but never used.

File: account-kit/smart-contracts/src/ma-v2/client/client.test.ts [hunk 2]
```diff
@@ -263,6 +264,105 @@ describe("MA v2 Tests", async () => {
...
```
Comment (moldy530) [inline account-kit/smart-contracts/src/ma-v2/client/client.test.ts:324]: ⚠️ **[eslint]** <[@typescript-eslint/no-unused-vars](https://typescript-eslint.io/rules/no-unused-vars)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'sendResult' is assigned a value but never used.

File: account-kit/smart-contracts/src/ma-v2/client/client.test.ts [hunk 3]
```diff
@@ -1,4 +1,10 @@
-import { custom, parseEther, publicActions, zeroAddress } from "viem";
+import {
+  custom,
+  getContract,
```
Comment (moldy530) [inline account-kit/smart-contracts/src/ma-v2/client/client.test.ts:3]: ⚠️ **[eslint]** <[@typescript-eslint/no-unused-vars](https://typescript-eslint.io/rules/no-unused-vars)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'getContract' is defined but never used.

File: account-kit/smart-contracts/src/ma-v2/client/client.test.ts [hunk 4]
```diff
@@ -14,10 +20,16 @@
 import { local070Instance } from "~test/instances.js";
 import { setBalance } from "viem/actions";
 import { accounts } from "~test/constants.js";
-import { getDefaultSingleSignerValidationModuleAddress } from "../modules/utils.js";
+import {
+  getDefaultSingleSignerValidationModuleAddress,
+  getDefaultAllowlistModuleAddress,
```
Comment (moldy530) [inline account-kit/smart-contracts/src/ma-v2/client/client.test.ts:25]: ⚠️ **[eslint]** <[@typescript-eslint/no-unused-vars](https://typescript-eslint.io/rules/no-unused-vars)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'getDefaultAllowlistModuleAddress' is defined but never used.

File: account-kit/smart-contracts/src/ma-v2/client/client.test.ts [hunk 5]
```diff
@@ -14,10 +20,16 @@
...
+  getDefaultNativeTokenLimitModuleAddress,
+} from "../modules/utils.js";
 import { SingleSignerValidationModule } from "../modules/single-signer-validation/module.js";
 import { allowlistModule } from "../modules/allowlist-module/module.js";
 import { HookType } from "../actions/common/types.js";
+import { nativeTokenLimitModule } from "../modules/native-token-limit-module/module.js";
+import { nativeTokenLimitModuleAbi } from "../modules/native-token-limit-module/abis/nativeTokenLimitModuleAbi.js";
```
Comment (moldy530) [inline account-kit/smart-contracts/src/ma-v2/client/client.test.ts:32]: ⚠️ **[eslint]** <[@typescript-eslint/no-unused-vars](https://typescript-eslint.io/rules/no-unused-vars)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'nativeTokenLimitModuleAbi' is defined but never used.

File: account-kit/smart-contracts/src/ma-v2/modules/native-token-limit-module/module.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import { encodeAbiParameters, type Address, type Hex } from "viem";
```
Comment (moldy530) [inline account-kit/smart-contracts/src/ma-v2/modules/native-token-limit-module/module.ts:1]: ⚠️ **[eslint]** <[@typescript-eslint/no-unused-vars](https://typescript-eslint.io/rules/no-unused-vars)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'Address' is defined but never used.

File: account-kit/smart-contracts/src/ma-v2/modules/native-token-limit-module/module.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+import { nativeTokenLimitModuleAbi } from "./abis/nativeTokenLimitModuleAbi.js";
```
Comment (moldy530) [inline account-kit/smart-contracts/src/ma-v2/modules/native-token-limit-module/module.ts:3]: ⚠️ **[eslint]** <[@typescript-eslint/no-unused-vars](https://typescript-eslint.io/rules/no-unused-vars)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'nativeTokenLimitModuleAbi' is defined but never used.

File: account-kit/smart-contracts/src/ma-v2/client/client.test.ts (hunk 4 above)
Comment (Zer0dot) [reply account-kit/smart-contracts/src/ma-v2/client/client.test.ts:25]: Will address in the relevant pr!

File: account-kit/smart-contracts/src/ma-v2/client/client.test.ts [hunk 8]
```diff
@@ -14,10 +14,15 @@ import {
 import { local070Instance } from "~test/instances.js";
 import { setBalance } from "viem/actions";
 import { accounts } from "~test/constants.js";
-import { getDefaultSingleSignerValidationModuleAddress } from "../modules/utils.js";
+import {
+  getDefaultSingleSignerValidationModuleAddress,
+  getDefaultAllowlistModuleAddress,
```
Comment (moldy530) [inline account-kit/smart-contracts/src/ma-v2/client/client.test.ts:19]: ⚠️ **[eslint]** <[@typescript-eslint/no-unused-vars](https://typescript-eslint.io/rules/no-unused-vars)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'getDefaultAllowlistModuleAddress' is defined but never used.

**Pull request merged**